unit-tests/algo/d2rgb/test-reproduction.cpp

```diff
@@ -209,16 +209,19 @@ int main( int argc, char * argv[] )
             try
             {
                 rs2_intrinsics raw_rgb_intr;
+                librealsense::algo::depth_to_rgb_calibration::rs2_intrinsics_double raw_rgb_intr_d;
                 read_binary_file( dir, "raw_rgb.intrinsics", &raw_rgb_intr );
 
+                raw_rgb_intr_d = raw_rgb_intr;
+                camera.rgb = raw_rgb_intr_d;
                 auto vec = read_vector_from< byte >( join( dir, "rgb_thermal_table" ) );
                 thermal::l500::thermal_calibration_table thermal_table( vec );
                 
                 auto scale = thermal_table.get_thermal_scale( settings.hum_temp );
                 AC_LOG( DEBUG, "Thermal {scale}" << scale << " [TH]" );
-                raw_rgb_intr.fx = float( raw_rgb_intr.fx * scale );
-                raw_rgb_intr.fy = float( raw_rgb_intr.fy * scale );
-                camera.rgb = raw_rgb_intr;
+                camera.rgb.fx = ( camera.rgb.fx * scale );
```

Comment:
The code here should match compare-scene, matlab, and `depth_to_rgb_calibration` -- so the fix should be everywhere.

It'd be nice if we can have one function that does the scaling and so enforces a 'rs2_intrinsics_double' type:

`rs2_intrinsics_double scale_intrinsics( rs2_intrinsics const &, float )`

It can go in AC utils?